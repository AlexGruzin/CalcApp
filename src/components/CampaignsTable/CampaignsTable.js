import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import classNames from 'classnames';

import './CampaignsTable.styles.less';

const CampaignsTable = ({ items, handleClick }) => {
  const { t } = useTranslation();

  return (
    <div className="table-cmp">
      <div className="table-cmp__description-row">
        <div className={classNames('table-cmp__label', 'description')}>
          <span>{t('Campaign')}</span>
        </div>
        <div className={classNames('table-cmp__type', 'description')}>
          <span>{t('Type')}</span>
        </div>
        <div className={classNames('table-cmp__part', 'description')}>
          <span>{t('Part of the Funnel')}</span>
        </div>
        <div className={classNames('table-cmp__status', 'description')}>
          <span>{t('Status')}</span>
        </div>
        <div className={classNames('table-cmp__spend', 'description')}>
          <span>{t('Spend')}</span>
        </div>
        <div className={classNames('table-cmp__roas', 'description')}>
          <span>{t('ROAS')}</span>
        </div>
      </div>

      {items.map(item => (
        <div className="table-cmp__row" onClick={handleClick}>
          <div className={classNames('table-cmp__label', 'item')}>
            <span>{item.label}</span>
          </div>
          <div className={classNames('table-cmp__type', 'item')}>
            <span>{item.type}</span>
          </div>
          <div className={classNames('table-cmp__part', 'item')}>
            <div className="able-cmp__parts">
              {item.part.map(part => (
                <span className={part}>{part}</span>
              ))}
            </div>
          </div>
          <div className={classNames('table-cmp__status', 'item')}>
            <span>{item.status}</span>
          </div>
          <div className={classNames('table-cmp__spend', 'item')}>
            <span>{item.spend}</span>
          </div>
          <div className={classNames('table-cmp__roas', 'item')}>
            <span>{item.roas}</span>
          </div>
        </div>
      ))}
    </div>
  );
};

CampaignsTable.propTypes = {
  items: PropTypes.array,
  handleClick: PropTypes.func,
};

CampaignsTable.defaultProps = {
  items: [],
  handleClick: () => {},
};

export default CampaignsTable;
