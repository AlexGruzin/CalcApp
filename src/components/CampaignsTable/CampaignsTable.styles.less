@import "~styles/generic.less";

.table-cmp {
  display: flex;
  flex-direction: column;
}

.table-cmp__description-row {
  display: flex;
  align-items: center;
  background: var(--purple);
  height: 35px;
  width: 100%;
  border-radius: 15px;
  margin-bottom: var(--space-x3);
}

.table-cmp__description-row span {
  .montserrat-regular();
  color: white;
}

.table-cmp__row {
  align-items: center;
  display: flex;
  background: var(--white);
  height: 35px;
  width: 100%;
  border-radius: 15px;
  margin-bottom: var(--space-x3);
  cursor: pointer;
}

.table-cmp__row:hover {
  background-color: rgba(200, 35, 176, 0.1);
}

.table-cmp__row span {
}

.table-cmp__label {
  width: 25%;
}

.table-cmp__type {
  width: 10%;
}

.table-cmp__part {
  width: 35%;
}

.table-cmp__status {
  width: 10%;
}

.table-cmp__spend {
  width: 10%;
}

.table-cmp__roas {
  width: 10%;
}

.table-cmp__row .table-cmp__label.item > span {
  .montserrat-bold();
  font-size: 14px;
  color: var(--grey700);
}

.table-cmp__row .item span {
  .montserrat-medium();
  font-size: 13px;
}

.table-cmp__label > span {
  margin-left: 40px;
}

.able-cmp__parts {
  display: flex;
}

.able-cmp__parts > span {
  display: flex;
  align-items: center;
  margin-left: var(--space-x2);
}

.able-cmp__parts > span:before {
  content: " ";
  display: inline-flex;

  height: 12px;
  width: 12px;
  border-radius: 50%;
  margin: 0 var(--space-x2);
}

.able-cmp__parts > span.Awareness:before {
  background: var(--purple);
}

.able-cmp__parts > span.Awareness:before {
  background: var(--turquoise);
}
