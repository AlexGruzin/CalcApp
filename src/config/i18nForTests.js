import i18n from 'i18next';

// locales
import US from '../locales/en';

i18n.init({
  debug: false,
  fallbackLng: ['US'],
  lng: 'US',

  ns: ['auth', 'resetPassword'],
  defaultNS: 'auth',

  resources: {
    US,
  },
});

export default i18n;

