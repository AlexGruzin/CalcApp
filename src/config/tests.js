import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next'
import i18n from "./i18nForTests";
import jsdom from 'jsdom';
const { JSDOM } = jsdom;

/*
 * jest-dom
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom/extend-expect';

/*
 * jest-fetch-mock
 * https://github.com/jefflau/jest-fetch-mock
 */
require('jest-fetch-mock').enableMocks();

if (global.document) {
  global.document.createRange = () => ({
    setStart: () => {
    },
    setEnd: () => {
    },
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
    createContextualFragment: str => JSDOM.fragment(str),
    // createContextualFragment: str => document.createElement('hr'),
  });
  global.open = () => undefined;
}

export const renderWithi18next = Component => {
  const Comp = React.cloneElement(Component, {
    changeLanguage: lng => {
      i18n.changeLanguage(lng);
      rerender(<I18nextProvider i18n={i18n}>{Comp}</I18nextProvider>);
    }
  });
  const defaultRender = render(
    <I18nextProvider i18n={i18n}>{Component}</I18nextProvider>
  );
  const { rerender } = defaultRender;
  return defaultRender;
};


// /**
//  * Wraps a component with an Intl Provider.
//  * @param {Object} component The component to render
//  */
// const withTestIntl = component => <IntlProvider locale="en">{component}</IntlProvider>;
//
// /*
//  * https://testing-library.com/docs/example-react-intl
//  */
// export const renderWithReactIntl = component => render(withTestIntl(component));

