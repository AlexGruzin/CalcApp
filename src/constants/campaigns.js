import { ADVERT, SOCIAL } from 'constants/advertisment';

const CAMPAIGNS_TYPES = {
  ECONOMY: 'ECONOMY',
  TEMP: 'TEMP',
};

export const CHART_HEIGHT = 300;

export const TYPES_ICONS = {
  [CAMPAIGNS_TYPES.ECONOMY]: 'leaf.svg',
  [CAMPAIGNS_TYPES.TEMP]: 'clock.svg',
};

export const CAMPAIGNS = [
  {
    label: 'Brand Loyalty',
    description: 'This campaign make sure that shoppers know about your brand',
    status: 1,
    type: CAMPAIGNS_TYPES.ECONOMY,
    icons: [ADVERT.INSTAGRAM, ADVERT.FACEBOOK, SOCIAL.BITMAP, SOCIAL.ENVELOPE],
  },
  {
    label: 'Category Awareness',
    description: 'This campaign focuses on making sure that shoppers who search for your categories sees your brand',
    status: 2,
    type: CAMPAIGNS_TYPES.ECONOMY,
    icons: [ADVERT.INSTAGRAM, ADVERT.FACEBOOK, SOCIAL.BITMAP, SOCIAL.ENVELOPE],
  },
  {
    label: 'Smart Shopping',
    description: 'Targets shoppers looking for your products on Google Shop',
    status: 2,
    type: CAMPAIGNS_TYPES.TEMP,
    icons: [ADVERT.INSTAGRAM, SOCIAL.BITMAP, SOCIAL.ENVELOPE],
  },
  {
    label: 'Re-Targeting',
    description: 'Remind visitors of your products and services after they leave your website without buying',
    status: 1,
    icons: [ADVERT.INSTAGRAM, ADVERT.FACEBOOK, SOCIAL.BITMAP],
  },
  {
    label: 'Competitor',
    description: 'Encourage visitors to come back to your website if we haven’t seen them in a while',
    status: 1,
    icons: [ADVERT.FACEBOOK, SOCIAL.BITMAP, SOCIAL.ENVELOPE],
  },
  {
    label: 'Re-Engagement',
    description: 'Encourage visitors to come back to your website if we haven’t seen them in a while',
    status: 3,
    type: CAMPAIGNS_TYPES.TEMP,
    icons: [ADVERT.FACEBOOK, SOCIAL.BITMAP, SOCIAL.ENVELOPE],
  },
  {
    label: 'Prospecting',
    description: 'Expand your reach to potential customers who will likely engage in your brand',
    status: 3,
    icons: [ADVERT.INSTAGRAM, ADVERT.FACEBOOK, SOCIAL.ENVELOPE],
  },
];

export const campaignsList = [
  {
    name: '1',
    label: 'Brand Loyalty',
    type: 'Evergreen',
    part: ['Awareness', 'Consideration'],
    status: 'active',
    spend: '$XXX.XX',
    roas: '$XXX.XX',
  },
  {
    name: '1',
    label: 'Category Awareness',
    type: 'Evergreen',
    part: ['Awareness', 'Consideration'],
    status: 'inactive',
    spend: '$XXX.XX',
    roas: '$XXX.XX',
  },
  {
    name: '1',
    label: 'Smart Shopping',
    type: 'Evergreen',
    part: ['Awareness', 'Consideration'],
    status: 'inactive',
    spend: '$XXX.XX',
    roas: '$XXX.XX',
  },
  {
    name: '1',
    label: 'Competition',
    type: 'Evergreen',
    part: ['Awareness', 'Consideration'],
    status: 'inactive',
    spend: '$XXX.XX',
    roas: '$XXX.XX',
  },
];

export const CAMPAIGN_TYPE = [
  {
    label: 'campaigns:campaignType.all',
    value: 0,
  },
  {
    label: 'campaigns:campaignType.evergreen',
    value: 1,
  },
  {
    label: 'campaigns:campaignType.holiday',
    value: 2,
  },
  {
    label: 'campaigns:campaignType.custom',
    value: 3,
  },
];

export const FUNNEL = [
  {
    label: 'campaigns:funnel.all',
    value: 0,
  },
  {
    label: 'campaigns:funnel.awareness',
    value: 1,
  },
  {
    label: 'campaigns:funnel.consideration',
    value: 2,
  },
  {
    label: 'campaigns:funnel.conversion',
    value: 3,
  },
];

export const STATUS = [
  {
    label: 'campaigns:status.all',
    value: 0,
  },
  {
    label: 'campaigns:status.active',
    value: 1,
  },
  {
    label: 'campaigns:status.inactive',
    value: 2,
  },
  {
    label: 'campaigns:status.draft',
    value: 3,
  },
  {
    label: 'campaigns:status.archived',
    value: 4,
  },
];

export const ORDER_BY = [
  {
    label: 'campaigns:orderBy.az',
    value: 0,
  },
  {
    label: 'campaigns:orderBy.za',
    value: 1,
  },
  {
    label: 'campaigns:orderBy.newestFirst',
    value: 2,
  },
  {
    label: 'campaigns:orderBy.longestRunningFirst',
    value: 3,
  },
  {
    label: 'campaigns:orderBy.mostProfitableFirst',
    value: 4,
  },
  {
    label: 'campaigns:orderBy.leastProfitableFirst',
    value: 5,
  },
];

export const NEXT_MONTH_ITEMS = [
  {
    title: 'Early Launch Black Friday Door Busters',
    subtitle: 'Needs images, headlines, and landing pages',
  },
  {
    title: 'The Great Reveal',
    subtitle: 'Needs videos and landing page',
  },
  {
    title: '25% OFF NEW & SALE SPEND PLUS $150 OR MORE 2) 25% OFF NEW & SALE SPEND PLUS $150 OR MORE 2)',
    subtitle: 'Needs images, headlines, and landing pages',
  },
  {
    title: 'VIP BLACK FRIDAY',
    subtitle: 'Needs images, headlines, and landing pages',
  },
  {
    title: 'Black Friday Sale',
    subtitle: 'Needs images, headlines, and landing pages',
  },
  {
    title: 'Cyber Monday Sale',
    subtitle: 'Needs videos and landing page',
  },
  {
    title: '30% off site wide no exclusions',
    subtitle: 'Needs images, headlines, and landing pages',
  },
  {
    title: 'GWP Eyewear',
    subtitle: 'Needs images, headlines, and landing pages',
  },
];

export const campaignsChartOptions = {
  data: {
    datasets: [
      {
        backgroundColor: '#7349BE',
        barThickness: 30, // weight line in px
        data: [10, 10, 10, 700, 10, 100, 470, 20],
        label: 1,
      },
      {
        backgroundColor: 'rgba(207, 39, 185, 1)',
        barThickness: 30, // weight line in px
        data: [200, 20, 20, 600, 40, 180, 900, 100],
        label: 2,
      },
    ],
    xLabels: [
      '10-Brand',
      '11-Brand-Category',
      '14-Brand-Coupon',
      '16-Brand-International',
      '17-Brand-Broad',
      '25-DSA',
      '55-Showcase-Shopping-Brand',
      '58-Smart-Shopping',
    ],
  },
  options: {
    responsive: true,
    maintainAspectRatio: false,
    legend: {
      display: false,
    },
    scales: {
      yAxes: [
        {
          gridLines: {
            color: 'rgba(0, 0, 0, 0)',
            drawTicks: false,
          },
          ticks: {
            callback(value) {
              return `$${value}`;
            },
            fontSize: 8,
            padding: 10,
          },
        },
      ],
      xAxes: [
        {
          gridLines: {
            drawTicks: false,
            borderDashOffset: 7, // why not working?
          },
          ticks: {
            fontSize: 8,
            padding: 10,
          },
        },
      ],
    },
  },
};
