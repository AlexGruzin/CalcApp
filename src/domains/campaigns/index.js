import AllActionsTypes from './actionTypes';
import * as AllActionsCreators from './actions';
import reducer from './reducer';

export { AllActionsTypes as ActionsTypes, AllActionsCreators as ActionsCreators, reducer };
