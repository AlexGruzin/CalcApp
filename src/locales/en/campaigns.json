{
  "campaignType": {
    "all": "All",
    "evergreen": "Evergreen",
    "holiday": "Holiday",
    "custom": "Custom"
  },
  "funnel": {
    "all": "All",
    "awareness": "Awareness",
    "consideration": "Consideration",
    "conversion": "Conversion"
  },
  "status": {
    "all": "All",
    "active": "Active",
    "inactive": "Inactive",
    "draft": "Draft",
    "archived": "Archived"
  },
  "title": "The dewy decimal system is to books as",
  "pinkTitle": "campaigns are to ads.",
  "campaigns": "Campaigns",
  "schedule": "Campaign Schedule",
  "filterBy": "Filter by",
  "dateRangeLabel": "Date Range",
  "campaignTypeLabel": "Campaign Type",
  "funnelLabel": "Funnel",
  "statusLabel": "Status",
  "live": "Live Campaigns are Live!",
  "orderByTitle": "Order By",
  "orderBy": {
    "az": "A-Z",
    "za": "Z-A",
    "newestFirst": "NewestFirst",
    "longestRunningFirst": "LongestRunningFirst",
    "mostProfitableFirst": "MostProfitableFirst",
    "leastProfitableFirst": "LeastProfitableFirst"
  },
  "table": {
    "startDate": "Start Date",
    "endDate": "End Date",
    "holiday": "Holiday",
    "campaign": "Campaign",
    "type": "Type",
    "LYRevenue": "LY Revenue",
    "goalRevenue": "Goal Revenue",
    "LYSpend": "LY Spend",
    "spendBudget": "Spend Budget"
  }
}