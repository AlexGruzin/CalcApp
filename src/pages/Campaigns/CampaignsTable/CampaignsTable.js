import React from 'react';
import PropTypes from 'prop-types';
import DataTable from 'react-data-table-component';
import { COLUMNS, TABLE_HEIGHT } from 'constants/campaignsTable';

const CampaignsTable = ({ allCampaigns }) => (
  <DataTable
    columns={COLUMNS}
    data={allCampaigns}
    striped
    noHeader
    className="campaingsTable"
    fixedHeaderScrollHeight={TABLE_HEIGHT}
  />
);

CampaignsTable.propTypes = {
  allCampaigns: PropTypes.array.isRequired,
};

CampaignsTable.defaultProps = {};

export default CampaignsTable;
