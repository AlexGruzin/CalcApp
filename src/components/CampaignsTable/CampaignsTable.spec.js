import React from 'react';
import { render } from '@testing-library/react';
import { campaignsList } from 'constants/campaigns';

import { renderWithi18next } from 'config/tests';

import CampaignsTable from './CampaignsTable';

describe('CampaignsTable tests', () => {
  it('Check snapshots of CampaignsTable are equal.', () => {
    const { asFragment } = renderWithi18next(<CampaignsTable items={campaignsList} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
