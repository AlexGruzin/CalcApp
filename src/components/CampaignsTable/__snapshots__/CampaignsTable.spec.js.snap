// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignsTable tests Check snapshots of CampaignsTable are equal. 1`] = `
<DocumentFragment>
  <div
    class="table-cmp"
  >
    <div
      class="table-cmp__description-row"
    >
      <div
        class="table-cmp__label description"
      >
        <span>
          Campaign
        </span>
      </div>
      <div
        class="table-cmp__type description"
      >
        <span>
          Type
        </span>
      </div>
      <div
        class="table-cmp__part description"
      >
        <span>
          Part of the Funnel
        </span>
      </div>
      <div
        class="table-cmp__status description"
      >
        <span>
          Status
        </span>
      </div>
      <div
        class="table-cmp__spend description"
      >
        <span>
          Spend
        </span>
      </div>
      <div
        class="table-cmp__roas description"
      >
        <span>
          ROAS
        </span>
      </div>
    </div>
    <div
      class="table-cmp__row"
    >
      <div
        class="table-cmp__label item"
      >
        <span>
          Brand Loyalty
        </span>
      </div>
      <div
        class="table-cmp__type item"
      >
        <span>
          Evergreen
        </span>
      </div>
      <div
        class="table-cmp__part item"
      >
        <div
          class="able-cmp__parts"
        >
          <span
            class="Awareness"
          >
            Awareness
          </span>
          <span
            class="Consideration"
          >
            Consideration
          </span>
        </div>
      </div>
      <div
        class="table-cmp__status item"
      >
        <span>
          active
        </span>
      </div>
      <div
        class="table-cmp__spend item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
      <div
        class="table-cmp__roas item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
    </div>
    <div
      class="table-cmp__row"
    >
      <div
        class="table-cmp__label item"
      >
        <span>
          Category Awareness
        </span>
      </div>
      <div
        class="table-cmp__type item"
      >
        <span>
          Evergreen
        </span>
      </div>
      <div
        class="table-cmp__part item"
      >
        <div
          class="able-cmp__parts"
        >
          <span
            class="Awareness"
          >
            Awareness
          </span>
          <span
            class="Consideration"
          >
            Consideration
          </span>
        </div>
      </div>
      <div
        class="table-cmp__status item"
      >
        <span>
          inactive
        </span>
      </div>
      <div
        class="table-cmp__spend item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
      <div
        class="table-cmp__roas item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
    </div>
    <div
      class="table-cmp__row"
    >
      <div
        class="table-cmp__label item"
      >
        <span>
          Smart Shopping
        </span>
      </div>
      <div
        class="table-cmp__type item"
      >
        <span>
          Evergreen
        </span>
      </div>
      <div
        class="table-cmp__part item"
      >
        <div
          class="able-cmp__parts"
        >
          <span
            class="Awareness"
          >
            Awareness
          </span>
          <span
            class="Consideration"
          >
            Consideration
          </span>
        </div>
      </div>
      <div
        class="table-cmp__status item"
      >
        <span>
          inactive
        </span>
      </div>
      <div
        class="table-cmp__spend item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
      <div
        class="table-cmp__roas item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
    </div>
    <div
      class="table-cmp__row"
    >
      <div
        class="table-cmp__label item"
      >
        <span>
          Competition
        </span>
      </div>
      <div
        class="table-cmp__type item"
      >
        <span>
          Evergreen
        </span>
      </div>
      <div
        class="table-cmp__part item"
      >
        <div
          class="able-cmp__parts"
        >
          <span
            class="Awareness"
          >
            Awareness
          </span>
          <span
            class="Consideration"
          >
            Consideration
          </span>
        </div>
      </div>
      <div
        class="table-cmp__status item"
      >
        <span>
          inactive
        </span>
      </div>
      <div
        class="table-cmp__spend item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
      <div
        class="table-cmp__roas item"
      >
        <span>
          $XXX.XX
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
